src/main/java/by/kirill/pympproject/service/ServiceProvider.java

```diff
@@ -13,10 +19,12 @@ public static ServiceProvider getInstance() {
     }
 
     public UserService getUserService() {
+        userService.setUserDao(userDao);
```

Comment:
In this case you will set `userDao` to the `userService` each time you call `getUserService()`. Can you do it just once? For ex. using initialization block?